server/src/main/java/jass/server/util/GameUtil.java

```diff
@@ -0,0 +1,160 @@
+     * @param clientPlayerFour  Player four.
+     */
+    public GameUtil(final ClientUtil clientPlayerOne, final ClientUtil clientPlayerTwo, final ClientUtil clientPlayerThree, final ClientUtil clientPlayerFour) {
+        this.clientPlayerOne = clientPlayerOne;
+        UserEntity playerOne = clientPlayerOne.getUser();
+        this.clientPlayerTwo = clientPlayerTwo;
+        UserEntity playerTwo = clientPlayerTwo.getUser();
+        this.clientPlayerThree = clientPlayerThree;
+        UserEntity playerThree = clientPlayerThree.getUser();
+        this.clientPlayerFour = clientPlayerFour;
+        UserEntity playerFour = clientPlayerFour.getUser();
+
+        // Add event listeners
+        clientPlayerOne.addChosenGameModeEventListener(this);
+
+        // Assign and create Teams
+        TeamEntity teamOne = new TeamEntity(playerOne, playerThree);
+        TeamRepository.getSingleton(null).add(teamOne);
+        TeamEntity teamTwo = new TeamEntity(playerTwo, playerFour);
+        TeamRepository.getSingleton(null).add(teamTwo);
+
+        // Initialize new Game
+        game = new GameEntity(teamOne, teamTwo, true);
+        GameRepository.getSingleton(null).add(game);
+        logger.info("Successfully created game with ID: " + game.getId());
+
+        // Broadcast to all Players that a game was created for them
+        GameFound gameFound = new GameFound(new GameFoundData(
+            game.getId(),
+            playerOne.getId(), playerOne.getUsername(), getTeamIdForPlayer(playerOne, teamOne, teamTwo),
+            playerTwo.getId(), playerTwo.getUsername(), getTeamIdForPlayer(playerTwo, teamOne, teamTwo),
+            playerThree.getId(), playerThree.getUsername(), getTeamIdForPlayer(playerThree, teamOne, teamTwo),
+            playerFour.getId(), playerFour.getUsername(), getTeamIdForPlayer(playerFour, teamOne, teamTwo)
+        ));
+        broadcast(gameFound);
+
+        // Create a round
+        CardUtil cardUtil = ServiceLocator.get(CardUtil.class);
+        RoundEntity newRound = new RoundEntity(playerOne, game);
+        RoundRepository.getSingleton(null).add(newRound);
+
+        // Send a deck to each user
+        List<DeckEntity> decks = cardUtil.addDecksForPlayers(newRound, playerOne, playerTwo, playerThree, playerFour);
+        cardUtil.broadcastDeck(clientPlayerOne, decks.get(0));
+        cardUtil.broadcastDeck(clientPlayerTwo, decks.get(1));
+        cardUtil.broadcastDeck(clientPlayerThree, decks.get(2));
+        cardUtil.broadcastDeck(clientPlayerFour, decks.get(3));
+
+        // Send a message to player one to choose a game mode
+        ChooseGameMode chooseGameMode = new ChooseGameMode(new ChooseGameModeData());
+        clientPlayerOne.send(chooseGameMode);
+    }
+
+    @Override
+    public void onChosenGameMode(final ChosenGameModeData data) {
+        // Check with token if player one actually is the one who sent the data.
+        if (data.getToken().equals(clientPlayerOne.getToken())) {
```

Comment:
in round 2 clientPlayerOne gets to pick the game mode. This case needs to be handled too.